rust/src/arrow/record_batch.rs

```diff
@@ -45,6 +45,10 @@ impl RecordBatchBuffer {
         self.batches.iter().map(|b| b.num_rows()).sum()
     }
 
+    pub fn push(&mut self, batch: RecordBatch) {
```

Comment:
should we make both mutable and immutable versions of a RecordBatchBuffer? or you think that's overkill?